opencl-aot/source/main.cpp

```diff
@@ -414,7 +414,14 @@ int main(int Argc, char *Argv[]) {
   CLProgramUPtr ProgramUPtr(std::move(Progs[0]));
 
   // step 7: set OpenCL build options
-  std::string BuildOptions = OptBuildOptions;
+  std::string BuildOptsStr;
+  if (!OptBuildOptions.empty()) {
+    BuildOptsStr = *OptBuildOptions.begin();
```

Comment:
Why this can't be handled within a loop below?

Reply:
It's done to avoid trailing space delimiter. If we start for_each from the first element, there will be a space at the beginning of the line.
Possibly, it's not a big problem, but I was not sure.